src/python_testing/TC_EWATERHTR_2_2.py

```diff
@@ -0,0 +1,378 @@
+                     "Verify value is 1 (Active)"),
+            TestStep("19", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER for Water Temperature 66C Test Event.",
+                     "Verify Command response is Success"),
+            TestStep("19a", "TH reads HeatDemand attribute.",
+                     "Verify value is 0x00 (no demand on any source)"),
+            TestStep("19b", "TH reads BoostState attribute.",
+                     "Verify value is 1 (Active)"),
+            TestStep("20", "TH sends Boost with Duration=600s,TemporarySetpoint=70C.",
+                     "Verify Command response is Success"),
+            TestStep("20a", "TH reads HeatDemand attribute.",
+                     "Verify value is greater than 0x00 (demand on at least one source) and (HeaterDemand & (!HeaterTypes)) is zero (demand is only from declared supported types)"),
+            TestStep("20b", "TH reads BoostState attribute.",
+                     "Verify value is 1 (Active)"),
+            TestStep("21", "TH sends CancelBoost.",
+                     "Verify Command response is Success"),
+            TestStep("21a", "TH reads HeatDemand attribute.",
+                     "Verify value is 0x00 (no demand on any source)"),
+            TestStep("21b", "TH reads BoostState attribute.",
+                     "Verify value is 0 (Inactive)"),
+            TestStep("22", "TH sends CancelBoost.",
+                     "Verify Command response is Success"),
+            TestStep("23", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to PIXIT.EWATERHTR.TEST_EVENT_TRIGGER for Basic installation Test Event Clear.",
+                     "Verify Command response is Success"),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EWATERHTR_2_2(self):
+
+        self.step("1")
+        # Commission DUT - already done
+
+        self.step("2")
+        await self.check_test_event_triggers_enabled()
+
+        self.step("3")
+        await self.send_test_event_trigger_basic_installation_test_event()
+
+        self.step("3a")
+        await self.check_whm_attribute("HeatDemand", 0)
+
+        self.step("3b")
+        await self.check_whm_attribute("BoostState", Clusters.WaterHeaterManagement.Enums.BoostStateEnum.kInactive)
+
+        self.step("3c")
+        heaterTypes = await self.read_whm_attribute_expect_success(attribute="HeaterTypes")
+        asserts.assert_greater(heaterTypes, 0,
+                               f"Unexpected HeaterTypes value - expected {heaterTypes} > 0")
+        asserts.assert_less_equal(heaterTypes, Clusters.WaterHeaterManagement.Bitmaps.WaterHeaterTypeBitmap.kOther,
```

Comment:
Is this less_equal used to confirm that the value is of the right type?

Reply:
No it's intended to check that the value is in the valid enum range since the test plan says check it is less than 0x20, but here kOther is 0x1F hence <= kOther, instead of <0x20.

It could be changed to < kUnsupportedValue if it was CPP, but in python that value isn't generated so we have to use <= kOther.

Again not sure if the underlying python type checking would ensure its a valid value, but suggest keeping as is because it shows that the test script and test plan are sort of in alignment - perhaps the test plan should say <= 0x1F.